### ISSUE TITLE
Building branch dev-mips-kitkat-v2013.10 fails

### ISSUE THREAD

**Issue by sfo** [MIPS/CI20_u-boot#3]
The build of branch dev-mips-kitkat-v2013.10 fails, so it is not possible to create an Android SD Card.
I manually retraced the respective commands from the mksdcard script (see CI20_android repo) and got the following output: http://pastebin.com/Dqb2BMfx

My procedure:
- add [...]/android/system/prebuilts/gcc/linux-x86/mips/mipsel-linux-android-4.6/bin to Path (also tried 4.7)
- set CROSS_COMPILE to mipsel-linux-android
- make distclean
- make ci20_mmc_config
- make

**Comment by chrisdearman** [comment]
The problem seems to be building u-boot-spl (image is too large). It's possible the relocation failures are a consequence of this. @crlarsen-it went through the complete build from github sources without problems AFAIK.

Clutching at straws here... could you try building in an en_US locale? I don't know if this could affect the u-boot build, but it is one obvious difference

**Comment by sfo** [comment]
I tried it with LC_ALL=C and LC_ALL=en_US on my Ubuntu 14.04.1 (configured german) machine and also on a clean ArchLinux (US english, i686 and x86_64) Live System, all with the same results.

**Comment by ZubairLK** [comment]
note sure. but perhaps the toolchain is making the slight difference of a few bytes?

**Comment by crlarsen-it** [comment]
My PATH environment variable includes ~/MentorGraphics/Sourcery_CodeBench_Lite_for_MIPS_GNU_Linux/bin

After running 'lunch aosp_ci20-userdebug' the following info was added to the PATH variable:
- /usr/lib/jvm/java-6-sun/bin
- ~/src/mipsandroid2/out/host/linux-x86/bin
- ~/src/mipsandroid2/prebuilts/gcc/linux-x86/mips/mipsel-linux-android-4.7/bin
- ~/src/mipsandroid2/development/emulator/qtools
- ~/src/mipsandroid2/prebuilts/gcc/linux-x86/mips/mipsel-linux-android-4.7/bin*
- ~/src/mipsandroid2/development/scripts
- ~/src/mipsandroid2/prebuilts/devtools/tools

*Yes, this path is really added twice.

CROSS_COMPILE never gets defined in my command shell, though it may get defined while some of the build procedures are being run.

I'm reworking my way through the build steps again to see if there's anything else which I missed in writing the directions.

**Comment by crlarsen-it** [comment]
I should apologize that there was a dependency which I had forgotten about when I wrote the directions. I've been using the tools so long that I simply forgot they're a part of my build environment. @chrisdearman pointed out to me that the web site does direct readers to the CodeBench toolchain.  I have updated the directions for building the Android image to refer the reader to the URL which explains where to find the tools, and how to install them.

**Comment by sfo** [comment]
Yes, it was an issue with the toolchain. Using the linked one works like a charm.
Thanks a lot :)

**Comment by jaretcantu** [comment]
While waiting for my ginormous other toolchain to download, I used the power of The Google to uncover [this nearly decade old gem](http://www.linux-mips.org/archives/linux-mips/2006-04/msg00188.html) (!):

> PIC code.  Use -fno-pic -mno-abicalls.
> Ralf

And now the problem: -mabicalls is unconditionally added to the PLATFORM_CPPFLAGS in arch/mips/config.mk; -fpic is applied conditionally depending on whether we are building the SPL or not. So -mabicalls should do the same (since -fpic requires -mabicalls).
I don't really see how to do this with build options, unless you want to run two separate makes with different flags -- one for the SPL and one for the regular u-boot.

I found it easier just to patch the mips/config.mk: 32476365eeea71e455b6d6da5ceb401ff2b6a5d8

I boot-tested with the 4.7 Android gcc, and it worked like a champ.
